sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/VbcHelper.cs

```diff
@@ -43,16 +43,16 @@ public static class VbcHelper
         /// </returns>
         public static bool IsTriggeringVbcError(Diagnostic diagnostic)
         {
-            var rootNode = diagnostic.Location?.SourceTree?.GetRoot();
-
-            if (rootNode != null &&
-                rootNode.Language == LanguageNames.VisualBasic &&
-                IsTextMatchingVbcErrorPattern(rootNode.FindNode(diagnostic.Location.SourceSpan)?.ToString()))
```

Comment:
This was just finding the node on which we were reporting the issue but that doesn't mean this is the node containing the error pattern... We need to look at the full line.